### ISSUE TITLE
Options for DOS commands are not preserved

### ISSUE THREAD

**Issue by Peter-A-Schneider** [msys2/msys2#64]
I have a build script which calls "cmd /d /c PATH", expecting the Windows cmd program to exit after displaying the PATH as Windows sees it. Unfortunately, msys interpretes the options as path names and accordingly transforms them to Windows drives. As a result the options are lost (and in this case, cmd does not return and the script hangs).

As much as the path conversion is normally desired: Is there a chance to add heuristics at least for built-in DOS commands (and cmd proper) to leave single letter options untouched? Possibly only known options? Probably only up to the first none-option argument, so that actually passing the path /c is not impossible? The set of these exceptions is clearly finite.

(I'm aware that I could escape the options, either with a forward or with a backslash. But that in turn would either involve relatively elaborate run time queries (in order to tell whether I'm running in msys2), or lock me in to msys2 because the options would then be wrong in other environments.)

**Comment by elieux** [comment]
> I have a build script which calls "cmd /d /c PATH"

What kind of a build script is this? It expects `cmd` to work, but doesn't expect path conversion? Where do you usually run it (Cygwin? WINE?)?

> The set of these exceptions is clearly finite.

Even if someone would be willing to write down the exceptions, there's little guarantee that a program called `cmd.exe` is always the one you think it is.

Anyway, would it help you to know that setting `MSYS2_ARG_CONV_EXCL=*` in the environment will disable argument conversions?

**Comment by Peter-A-Schneider** [comment]
@elieux 
Yes, it currently runs under cygwin but we interface heavily with Windows programs. Msys could be nice there, mostly because of the path conversion, paradoxically.

And true: cmd is not necessarily Windows'; in fact it is /bin/cmd. I now use that script as a hook to escape single letter options, but that is not a general solution.

MSYS2_ARG_CONV_EXCL: Yes, that helps, thank you. We are unlikely to have naked drive names as actual path arguments anywhere in our build scripts.

**Comment by elieux** [comment]
Great. I hope that MSYS2 will serve you well.

**Comment by Peter-A-Schneider** [comment]
I actually don't think the issue is solved by just setting MSYS2_ARG_CONV_EXCL, generally. Other people may have naked drive names as arguments, which is probably why they are not generally excluded from conversion. What I was suggesting was to improve the heuristics of the conversion algorithm, depending on the executable as well as the position and contents of the argument.

**Comment by elieux** [comment]
It doesn't seem realistic to me, for both the abovementioned reason and backwards compatibility concerns. I have to defer to @Alexpux anyway, so let's reopen and wait for his opinion (he's on vacation currently).
